compiler/passes/externCResolve.cpp

```diff
@@ -616,26 +596,44 @@ static Symbol* tryCResolve(BaseAST* context,
   return tryCResolveLocally(module, name);
 }
 
+Symbol* tryCResolve(ModuleSymbol* mod, const char* name) {
+  Symbol* retval = NULL;
+
+  // Try looking up the symbol with scope resolve's tables.
+  // This covers the case of finding a symbol already converted.
+  int nSymbolsFound = 0;
+  Symbol* got = lookupInModuleOrBuiltins(mod, name, nSymbolsFound);
+  if (nSymbolsFound != 0)
+    return got;
+
+  if (fAllowExternC == true) {
+    llvm::SmallSet<ModuleSymbol*, 24> visited;
+
+    retval = doTryCResolve(mod, name, visited);
+  }
+
+  return retval;
+}
+
 static void addCDef(ModuleSymbol* module, DefExpr* def) {
   module->block->insertAtHead(def);
 
   addToSymbolTable(def);
 }
 
 static Expr* tryCResolveExpr(ModuleSymbol* module, const char* name) {
-  BaseAST* context = module->block;
-  Symbol* sym = tryCResolve(context, astr(name));
+  Symbol* sym = tryCResolveLocally(module, astr(name));
   if (sym == NULL)
-    USR_FATAL(context, "Could not find C type %s", name);
+    USR_FATAL("Could not find C type %s", name);
```

Comment:
Doesn't this make the errors less good, by removing the location information of where the call that triggered the error lived?  I'm surprised there weren't any tests updated as a result of this change

Reply:
Yes but since this function is only used internally, I think the most likely cause is that there was an internal error within externCResolve. I have been trying not to stress too much about it though, because adding the context would interfere with the caching, and I am expecting to revisit this code in dyno soon.

Reply:
Oh, I see.  The module being passed in is not necessarily the most recent call?

Reply:
Right, I was concerned that tryCResolve would have different behavior in different contexts, but that shouldn't matter here (since this is a fatal error). When we call `tryCResolve` in `lookupThisScopeAndUses`, we could pass in `context` in addition to `scopeModule`, but then we would have to thread this `context` through a lot of the code in externCResolve, which IMO is more trouble than it is worth right now.

Anyway I'm not actually convinced the error previously would have been better, because we would have just been pointing the error to the first line of the module, which IMO is more confusing than useful.

Reply:
Maybe worth opening an "error message" future if there isn't one already, but I wouldn't hold up the PR to add it